src/CISystems.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    enum CI_SYSTEM
+
+The supported CI systems to deploy docs from, using Documenter.jl.
+
+Values:
+* [`TRAVIS`](@ref)
+* [`GITLAB_CI`](@ref)
+* [`CIRRUS_CI`](@ref)
+* [`DRONE`](@ref)
+* [`APPVEYOR`](@ref)
+"""
+@enum CI_SYSTEM begin
+    TRAVIS
+    GITLAB_CI
+    CIRRUS_CI
+    DRONE
+    APPVEYOR
+end
+# docstrings for CI_SYSTEM values
+"Specialize [`deploydocs`](@ref) for [Travis CI](https://travis-ci.org/). See also [`Documenter.CI_SYSTEM`](@ref)."
+TRAVIS
+"Specialize [`deploydocs`](@ref) for [Appveyor](https://www.appveyor.com/). See also [`Documenter.CI_SYSTEM`](@ref)."
+APPVEYOR
+"Specialize [`deploydocs`](@ref) for [GitLab CI](https://about.gitlab.com/product/continuous-integration/). See also [`Documenter.CI_SYSTEM`](@ref)."
+GITLAB_CI
+"Specialize [`deploydocs`](@ref) for [Cirrus CI](https://cirrus-ci.org/). See also [`Documenter.CI_SYSTEM`](@ref)."
+CIRRUS_CI
+"Specialize [`deploydocs`](@ref) for [Drone](https://drone.io/). See also [`Documenter.CI_SYSTEM`](@ref)."
+DRONE
+
+"""
+    read_ci_env([returnfinaldeploy::Bool]; uploader::CI_SYSTEM = TRAVIS)
+
+Read the CI environment variables, and return (always) a NamedTuple where the first
+five elements are, in order:
+    - `cibranch`: which branch is CI running on?
+    - `pull_request`: is CI building a pull request?  Empty string if not.
+    - `repo_slug`: The URL-friendly repo name.
+    - `tag`: The tag being built bu CI.  Empty string if not.
+    - `event_type`: The event that triggered CI.
+Furthermore, if `returnfinaldeploy` is true, the function will check whether the
+current CI provider matches `uploader` (set by default to Travis).
+
+Returns a NamedTuple.
```

Comment:
Maybe, for a more cleaner API here, it would be better if `read_ci_env` would just return `nothing` if the CI env can not be detected? That way the call sites do not have to do the `haskey(ENV, "CI")` check, which may actually be different on different CI systems (in general). So instead, they would check if this returned `nothing` or not, and decide whether we're in CI based on that.